### ISSUE TITLE
Browser Certificate Warnings

### ISSUE THREAD

**Issue by refringe** [laravel/homestead#526]
The latest beta version of Chrome (and apparently Firefox) make a change to how self-signed certificates are validated:  
https://www.chromestatus.com/feature/4981025180483584

Open any of your Homestead projects in the beta version of Chrome (58.0.3029.19) right now and (even with the SSL certificate imported into the system keychain and explicitly allowed) you will receive a `net::ERR_CERT_COMMON_NAME_INVALID` certificate error page.

More information about why *that* error is shown here:
https://bugs.chromium.org/p/chromium/issues/detail?id=308330

In short, it's no longer good enough to only use the commonName (CN) field for the domain(s) when generating a certificate. The subjectAltName must be used to list domains that the certificate validates. This would typically involve modifying the OpenSSL configuration file for each domain/certificate that you want to generate. 

How to do that properly is written about here:
http://stackoverflow.com/questions/21488845/how-can-i-generate-a-self-signed-certificate-with-subjectaltname-using-openssl

**Comment by svpernova09** [comment]
I'd rather not add multiple domains to the cert since I'd rather avoid such practices (even in dev) .

Will look into this a bit. I'm out of town until Thursday if someone wants to take a crack at this.

--
- Joe Ferguson
JoeFerguson.me
MemphisPHP.org

**Comment by refringe** [comment]
Understood. I've modified the initial issue and removed the suggestion. Practices aside, it would make detecting previously generated certificates (for regeneration purposes) a pain. More simple to keep them separate. 👍

**Comment by refringe** [comment]
I'll see if I can take a swing at it tonight.

**Comment by ryancwalsh** [comment]
@svpernova09 and @refringe I wonder if you know how to solve this Laravel Homestead local dev HTTPS issue: https://stackoverflow.com/q/48969083/470749 Thanks so much for your help!

**Comment by browner12** [comment]
we now generate a root certificate when the machine is built.  you can find it in `/etc/nginx/ssl`.  Simply copy it onto your host machine, and then make it a trusted certificate.  You can find instructions for your specific OS online.

**Comment by ryancwalsh** [comment]
@browner12 Thanks but I already tried that. (I read about it here: https://stackoverflow.com/a/47755133/470749)

**Comment by ryancwalsh** [comment]
@svpernova09 , @refringe , @browner12 I've spent so many hours over so many weeks on this Homestead SSL issue and haven't made any progress. I'd love any suggestions.  Both Windows 10 Chrome Version 65.0.3325.181 (Official Build) (64-bit) and Firefox Quantum 59.0.2 (64-bit) are saying that my Homestead HTTPS connection is not secure. I've tried using a `.rcw` TLD, the `.test` domain (as suggested [here](https://github.com/laravel/homestead/issues/754)), and the http://xip.io/ service.

The error in Chrome says:

> This page is not secure (broken HTTPS).
> Certificate - missing
> This site is missing a valid, trusted certificate (net::ERR_CERT_AUTHORITY_INVALID).

I've tried copying the root certificate from Homestead as suggested by @browner12 and [here](https://stackoverflow.com/a/47755133/470749).

If you could suggest anything I should look into at this point, I'd appreciate it. Thanks!

**Comment by svpernova09** [comment]
@ryancwalsh  Please do not spam every possible issue. From your SO post, you're not running the latest version of Homestead. The first suggestion I have is to update Homestead to the latest versions of the repo and the base box and then visit one of your sites and add an SSL exception and trust the certificate it provides. If you have any further issues please open a new issue and fill out the template completely.

**Comment by ryancwalsh** [comment]
@svpernova09 Thanks, Joe. I'd meant to update in my SO post that I'd already updated the versions. If I understand correctly, everything already *is* up to date. I've created a new issue (linked above a minute ago). Thanks so much for your help.
